### ISSUE TITLE
Request: Add total tile count or module state to AbstractTileProvider

### ISSUE THREAD

**Issue by DennyLindberg** [mapbox/mapbox-unity-sdk#897]
In relation to issue #896 I would like to request either functionality in AbstractTileProvider:

**Add current and total tile count**
This tile count should be the number of tiles currently produced by the tile provider. This would allow us to keep track of the current progress of the tile provider.

**Add module state to the tile provider**
As show in #896 we get some unintended behaviour in the code, which could be solved by keeping track of the state of the tile provider. By adding support for a working state we can circumvent other classes like AbstractMapVisualizer finishing too early.

**Comment by BergWerkGIS** [comment]
As mentioned in https://github.com/mapbox/mapbox-unity-sdk/issues/896#issuecomment-402093003 this is in the works (will need some time till it arrives in a release though).

One thing to be aware of and we are trying to integrate as smoothly as possible: there's a chance that the total tile count and state might be continuously changing eg if the user continuously pans around.

**Comment by DennyLindberg** [comment]
Which I think is perfectly fine. 

We wanted to minimize the amount of full-map updates we are doing on our end in regards to some custom paths that we implemented from an external data source. Even if the tiles are continuously loading from the tile provider we found it to be an improvement to wait for the whole group of tiles in the current frame to load rather than re-generate on each separate addition.
